src/os/bluestore/BlueStore.h

```diff
@@ -513,8 +513,9 @@ class BlueStore : public ObjectStore,
       ++nref;
     }
     void put() {
-      if (--nref == 0)
-	delete this;
+      if (--nref == 0) {
+	//delete this;
+      }
```

Comment:
> Shouldn't we do blob (partial?) cleanup on putting into the pool. Otherwise it might waste a huge amount of space while being on the bench...

Yup, this will be addressed together with removal of Blob's ref counting. At the moment the memory is freed with the whole pool. See  [the initial comment](https://github.com/ceph/ceph/pull/13982#discussion_r106192324).

> Another concern (applicable to both pools) - we return a 'dirty' object, i.e one that contains some information from the prev round - IMO this might cause some issues as post-allocation code might rely on some specific (clean) object state...

Not sure whether we're on the same page. I encountered `BlueStore::Blob::dirty_blob()` that returns a non-const reference to `bluestore_blob_t`. I can't find any place where such reference is buried inside other structures. Moreover, it shouldn't be as -- without some ugly hacks -- there is no possibility to manipulate `Blob`'s ref counter basing solely on it.